### DISCUSSION TITLE
Can I get the speed of my car using this mapbox library?

### DISCUSSION THREAD

**Discussion by defi-bear** [rnmapbox/maps#1664] [Q&A]
Hi guys,

I would like to get the speed of my car and get the speed limit of the road so I can show if the driver is driving the car very fast if the speed exceeds the limit.

Could you please teach me how can I get both info?
Thanks

**Comment by ferdicus** [comment]
Hi 👋🏿 

Speed is part of the Coordinates you get when the GPS fires a location update:
https://github.com/react-native-mapbox-gl/maps/blob/d6e7257e705b8e0be5d2d365a495c514b7f015f5/index.d.ts#L240-L243

So that's where you get the speed from - easy. 


Speed limit on the other hand is pretty tricky:
* get direction info from a service (Google Directions API/ HERE Directions API/ etc)
* within that direction info should be some form of line layer that describes the route to take
* no you have to snap the position you get from your GPS on that line
* on that section of the line - what is the speed limit? (does the service provider pack that info in there?)

That's how you get the speed limit - harder

**Reply by defi-bear** [reply]
Thank you for your reply. I just checked and I can get speed well. I will need to check more for speed limit. I will try the ways that you wrote and will back again. Thanks

**Reply by defi-bear** [reply]
Hey @ferdicus,

I still didn't understand about this solution. I was able to get the location and name by using mapbox direction api but how can I get the speed limit from it? Could you tell me more detaily?

So what I am going to do is showing the current speed and speed limit of that street when driver is driving through the street.

**Reply by ferdicus** [reply]
hi, well my solution was under the assumption, that your direction api payload includes a value for `speed_limit` or something similar.
I didn't check if mapbox' direction api actually provides that info... I know [GOOGLE](https://developers.google.com/maps/documentation/roads/speed-limits) and [HERE](https://developers.here.com/blog/getting-started-with-speed-limits-and-fleet-telematics) show speed limits
